rfcs/0124-worker-manager.md

```diff
@@ -0,0 +1,271 @@
+
+### Documentation and Payload Schemata
+Each worker type and worker configuration might have documentation and payloads
+associated with them.
+
+The schema will be exposed as any other value in the worker configuration to
+the JSON document as an option like any other, but will be treated as an opaque
+JSON-Schema document.  If overridden in a more specific configuration section,
+the most specific complete document will be used.  Accessing the payload schema
+by worker type will be supported.
+
+Documentation will be supported through a second API endpoint.  Documentation
+will be a markdown document.  No secondary documents will be stored along with
+the markdown.  If documentation requires images or other content, it can be
+embedded with data uris, or hosted externally.  Documentation can be provided
+for worker configurations as well as worker types.  No templating will be
+supported.
+
+Documentation can be requested by worker configuration, or by worker type.
+When requesting by worker configuration, only the worker configuration
+documentation will be shown.  When requesting documentation for a worker type,
+the worker type documentation will be show, then a header similar to "# General
+Worker Configuration Documentation" followed by the worker configuration
+documentation.
+
+## Provisioning with Worker Manager
+The worker manager will take over provisioning from the ad-hoc provisioners
+which exist today.  Provisioning can broadly be broken into two components:
+bidding and allocation.
+
+Bidding operates on Taskcluster concepts.  It takes into account the amount of
+pending work, the number of running workers, and worker configuration then
+decides how much capacity should be created.
+
+Allocation happens when workers need to be created.  It's the mechanics of how
+to create a worker for a given worker type.  Allocation operates purely on
+cloud provider concepts.
+
+The worker manager will have bidding strategies and providers.  The bidding
+strategies will provide different ways to decide whether new capacity is
+needed.  Providers will understand how to suggest bids for a given capacity
+requirement as well as allocating the selected bids.
+
+The bidding strategy will be configurable in the worker configuration by
+setting the identifier of the strategy for that worker configuration.
+Providers will be configured globally.  Each worker configuration will need to
+opt in to each provider as well as give provider specific configuration.
+
+### Bidding
+Bidding strategies will take as inputs the current worker type state, the
+recent history of a worker type, the worker configuration and the amount of
+work pending from the Queue.  They will calculate the number of capacity units
+which are needed.  If that number is greater than zero, they will request bids
+from the providers.
+
+The bid request will be the number of capacity units needed.  The bid response
```

Comment:
This might be another good place for an example block showing what a request might look like.

Reply:
I hope the conversational format I used is ok :)